docs/design/segments.md

```diff
@@ -143,6 +143,11 @@ the 'column data' is an array of values. Additionally, a row with *n*
 values in 'column data' will have *n* non-zero valued entries in
 bitmaps.
 
+## SQL Compatible Null Handling
+By default, Druid string dimension columns use the values `''` and `null` interchangeably and numeric and metric columns can not represent `null` at all, instead coercing nulls to `0`. However, Druid also provides an SQL compatible null handling mode, which must be enabled at the system level, through `druid.generic.useDefaultValueForNull`. This setting, when set to `false`, will allow Druid to _at ingestion time_ create segments whose string columns can distinguish `''` from `null`, and numeric columns which can represent `null` valued rows instead of `0`.
```

Comment:
I'd go with "a SQL" not "an SQL".

https://oracle-base.com/blog/2015/01/02/a-sql-or-an-sql/ presents some arguments for both sides, but the "an SQL" side is clearly misguided.

Reply:
![](https://media.giphy.com/media/zqhZB6bo5FgoE/giphy.gif)

but fine I will change all occurrences that I added 😛